webserver/views/datasets.py

```diff
@@ -92,6 +93,41 @@ def view(id):
     )
 
 
+def get_json(id):
+    dataset = get_dataset(id)
+    dataset_clean = {
+        "name": dataset["name"],
+        "description": dataset["description"],
+        "classes": [],
+        "public": dataset["public"],
+    }
+    for cls in dataset["classes"]:
+        dataset_clean["classes"].append({
+            "name": cls["name"],
+            "description": cls["description"],
+            "recordings": cls["recordings"],
+        })
+
+    return dataset_clean
+
+CSV_DIR = "./code/data/files/csv_files"
+
+@datasets_bp.route("/<uuid:dataset_id>/download")
+def download_csv(dataset_id):
+    # return render_template("datasets/accuracy.html")
+    ds = get_json(dataset_id)
+    ds_csv = ""
+    for ds_class in ds["classes"]:
+        class_name = ds_class["name"]
+        for rec in ds_class["recordings"]:
+            strList = [str(rec), str(class_name)]
+            ds_csv = ds_csv + ",".join(strList) + "\n"
```

Comment:
Constructing CSV by hand is always a bad idea. Use the [csv.writer](https://docs.python.org/2/library/csv.html#csv.writer) module. You can write to a [stringio](https://docs.python.org/2/library/stringio.html) for a "file-like object":

```py
fp = StringIO.StringIO()
writer = csv.writer(fp)
for class in classes:
   for recording in recordings:
        writer.writerow([recording, class])

fp.getvalue() # the resulting csv file
```